src/CoreShop/Component/Pimcore/Templating/Helper/LinkGeneratorHelper.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * CoreShop.
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2015-2017 Dominik Pfaffenbauer (https://www.pfaffenbauer.at)
+ * @license    https://www.coreshop.org/license     GNU General Public License version 3 (GPLv3)
+ */
+
+namespace CoreShop\Component\Pimcore\Templating\Helper;
+
+use CoreShop\Component\Pimcore\Routing\LinkGeneratorInterface;
+use Pimcore\Model\DataObject\Concrete;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Templating\Helper\Helper;
+
+class LinkGeneratorHelper extends Helper implements LinkGeneratorHelperInterface
+{
+    /**
+     * @var LinkGeneratorInterface
+     */
+    private $linkGenerator;
+
+    /**
+     * @param LinkGeneratorInterface $linkGenerator
+     */
+    public function __construct(LinkGeneratorInterface $linkGenerator)
+    {
+        $this->linkGenerator = $linkGenerator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPath($routeNameOrObject/*, $routeName*/, $params = [], $relative = false)
+    {
+        list ($object, $routeName, $params, $relative) = $this->prepareParameters(func_get_args());
+
+        return $this->linkGenerator->generate($object, $routeName, $params, $relative ? UrlGeneratorInterface::RELATIVE_PATH : UrlGeneratorInterface::ABSOLUTE_PATH);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUrl($routeNameOrObject/*, $routeName*/, $params = [], $schemeRelative = false)
+    {
+        list ($object, $routeName, $params, $relative) = $this->prepareParameters(func_get_args());
+
+        return $this->linkGenerator->generate($object, $routeName, $params, $relative ? UrlGeneratorInterface::NETWORK_PATH : UrlGeneratorInterface::ABSOLUTE_URL);
+    }
+
+    /**
+     * @param $arguments
+     * @return array
+     */
+    protected function prepareParameters($arguments)
+    {
```

Comment:
Extending the default symfony helper class so we don't neeed to add this method?

Reply:
this method allows you to do following thing:

```twig
coreshop_path($object, 'route', {}, false)
//and
coreshop_path('route', {}, false)
//and
coreshop_path($object)
//and
coreshop_path('route_name')
```

So, it checks what kind of arguments come and therefore we don't need to call the function without a `null`parameter for non object-routes.